File: frontend/src/api/jobApplication.ts
/**
 * 高校教学就业平台 - 求职申请相关 API
 */
import request from '@/utils/request'

// 求职申请接口
export interface JobApplication {
  id?: number
  studentId?: number
  studentName?: string
  positionId?: number
  positionName?: string
  companyId?: number
  companyName?: string
  // 状态: pending-待处理 screened-已筛选 test_passed-笔试通过
  // test_failed-笔试失败 interview_passed-面试通过 interview_failed-面试失败
  // rejected-已拒绝 offered-已发offer hired-已入职 declined-已拒绝offer
  status?: string
  // 当前阶段: resume-简历筛选 test-笔试 interview-面试 offer-offer发放 hired-入职
  currentStage?: string
  resumeId?: number
  applicationTime?: string
  hrRemark?: string
  createTime?: string
  updateTime?: string
}

// 分页查询参数
export interface JobApplicationPageParams {
  current: number
  size: number
  positionId?: number
  studentId?: number
  companyId?: number
  status?: string
  currentStage?: string
}

// 分页结果
export interface JobApplicationPageResult {
  records: JobApplication[]
  total: number
  current: number
  size: number
}

/**
 * 分页查询求职申请列表
 */
export function getJobApplicationPageApi(params: JobApplicationPageParams): Promise<JobApplicationPageResult> {
  return request({
    url: '/job-application/page',
    method: 'get',
    params
  }) as Promise<JobApplicationPageResult>
}

/**
 * 根据ID查询求职申请
 */
export function getJobApplicationByIdApi(id: number) {
  return request({
    url: `/job-application/${id}`,
    method: 'get'
  })
}

/**
 * 更新求职申请状态
 */
export function updateJobApplicationStatusApi(id: number, status: string, remark?: string) {
  return request({
    url: `/job-application/${id}/status`,
    method: 'put',
    data: { status, remark }
  })
}

/**
 * 批量更新求职申请状态
 */
export function batchUpdateJobApplicationStatusApi(ids: number[], status: string) {
  return request({
    url: '/job-application/batch-status',
    method: 'put',
    data: { ids, status }
  })
}

/**
 * 删除求职申请
 */
export function deleteJobApplicationApi(id: number) {
  return request({
    url: `/job-application/${id}`,
    method: 'delete'
  })
}

/**
 * 批量删除求职申请
 */
export function batchDeleteJobApplicationApi(ids: number[]) {
  return request({
    url: '/job-application/batch',
    method: 'delete',
    data: { ids }
  })
}

/**
 * 新增求职申请
 */
export function createJobApplicationApi(data: JobApplication) {
  return request({
    url: '/job-application',
    method: 'post',
    data
  })
}

/**
 * 更新求职申请
 */
export function updateJobApplicationApi(data: JobApplication) {
  return request({
    url: '/job-application',
    method: 'put',
    data
  })
}

/**
 * 筛选简历
 */
export function screenResumeApi(id: number, passed: boolean, remark?: string) {
  return request({
    url: `/job-application/${id}/screen`,
    method: 'post',
    params: { passed, remark }
  })
}

/**
 * 更新申请状态和阶段
 */
export function updateApplicationStatusApi(
  id: number,
  status: string,
  currentStage: string
) {
  return request({
    url: `/job-application/${id}/update-status`,
    method: 'post',
    params: { status, currentStage }
  })
}

/**
 * 撤销申请
 */
export function withdrawApplicationApi(id: number, studentId: number) {
  return request({
    url: `/job-application/${id}/withdraw`,
    method: 'post',
    params: { studentId }
  })
}

File: backend/src/main/java/com/teaching/employment/entity/JobApplication.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 求职申请实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-12
 */
@Data
@TableName("t_job_application")
@ApiModel(description = "求职申请信息")
public class JobApplication implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "申请ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "职位ID")
    private Long positionId;

    @ApiModelProperty(value = "学生ID")
    private Long studentId;

    @ApiModelProperty(value = "简历ID")
    private Long resumeId;

    @ApiModelProperty(value = "申请状态：pending-待处理 screened-已筛选 test_passed-笔试通过 interview_passed-面试通过 test_failed-笔试失败 interview_failed-面试失败 rejected-已拒绝 offered-已发offer hired-已入职 declined-已拒绝offer")
    private String status;

    @ApiModelProperty(value = "当前阶段：resume-简历筛选 test-笔试 interview-面试 offer-offer发放 hired-入职")
    private String currentStage;

    @ApiModelProperty(value = "申请时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime applicationTime;

    @ApiModelProperty(value = "HR备注")
    private String hrRemark;

    @ApiModelProperty(value = "是否删除：1-已删除 0-未删除")
    @TableLogic
    private Integer isDeleted;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 职位信息（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private Position position;

    /**
     * 学生信息（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private Student student;

    /**
     * 简历信息（关联查询时使用，不映射到数据库字段）
     */
    @TableField(exist = false)
    private Resume resume;

    /**
     * 职位名称
     */
    @TableField(exist = false)
    private String positionName;

    /**
     * 学生姓名
     */
    @TableField(exist = false)
    private String studentName;

    /**
     * 学生电话
     */
    @TableField(exist = false)
    private String studentPhone;

    /**
     * 企业名称
     */
    @TableField(exist = false)
    private String companyName;
}
